cli/src/commit_templater.rs

```diff
@@ -2214,7 +2214,12 @@ impl TreeDiff {
 
     fn into_formatted<F, E>(self, show: F) -> TreeDiffFormatted<F>
     where
-        F: Fn(&mut dyn Formatter, &Store, BoxStream<CopiesTreeDiffEntry>) -> Result<(), E>,
+        F: Fn(
+            &mut dyn Formatter,
+            &Store,
+            Diff<&MergedTree>,
```

Comment:
Is `Diff<&ConflictLabels>` sufficient? That would seem clearer and less redundant, or what do you think?